universal-application-tool-0.0.1/app/controllers/applicant/ApplicantProgramBlocksController.java

```diff
@@ -258,25 +259,40 @@ public ApplicantProgramBlocksController(
 
               Optional<String> bucket = request.queryString("bucket");
               Optional<String> key = request.queryString("key");
+
+              // Original file name is only set for Azure, where we have to generate a UUID when
```

Comment:
Trial and error honestly. This has to be done because of the way the Azure Blob Storage SDK works, compared to how S3 does. Future contributors will have to make a decision based on how GCP file upload works, I believe

Reply:
For the record, GCS supports POST uploads so we won't have to do this weird workaround. Azure is the odd one out here. https://cloud.google.com/storage/docs/xml-api/post-object-forms

Reply:
Would that be worth adding as a comment for future contributors or is that over kill?

Reply:
> Would that be worth adding as a comment for future contributors or is that over kill?

the more context/info in the comments the better, honestly

Reply:
Added! Thanks for that info and link @shubha-rajan

Reply:
"Future contributors will have to make a decision based on how GCP file upload works"

How would they know they need to make that decision, or the decision to make?  

IE: Should we have a wikipage on FileUpload and include this?

You're in effect designing a general way that this works now, so the more we can set future folks up for success the better

Reply:
Yes!! This sounds like an excellent subject for docs day I'll add it to the list